clang/test/CodeGen/cheri/intcap-const-expr-casts.c

```diff
@@ -0,0 +1,48 @@
+// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py UTC_ARGS: --check-globals --include-generated-funcs --global-value-regex ".+"
+// RUN: %riscv64_cheri_cc1 -o - -emit-llvm -Wno-cheri-capability-misuse %s | FileCheck --check-prefix=HYBRID %s
+// RUN: %riscv64_cheri_purecap_cc1 -o - -emit-llvm -Wno-cheri-capability-misuse %s | FileCheck --check-prefix=PURECAP %s
+
+extern char x[];
+
+/// These should all preserve provenance
+__intcap s = (__intcap)x;
+unsigned __intcap u = (unsigned __intcap)x;
+__intcap u2s = (__intcap)(unsigned __intcap)x;
+unsigned __intcap s2u = (unsigned __intcap)(__intcap)x;
+void * __capability s2p = (void * __capability)(__intcap)x;
```

Comment:
Clang-format will be disabled for test/ when we merge next anyway, but LLVM style is `int *const` and therefore `int *__capability` (although the codebase is quite inconsistent).